pkg/controllers/status/common.go

```diff
@@ -105,43 +108,63 @@ func updateResourceStatus(
 	dynamicClient dynamic.Interface,
 	restMapper meta.RESTMapper,
 	interpreter resourceinterpreter.ResourceInterpreter,
-	resource *unstructured.Unstructured,
+	eventRecorder record.EventRecorder,
+	objRef workv1alpha2.ObjectReference,
 	bindingStatus workv1alpha2.ResourceBindingStatus,
 ) error {
-	gvr, err := restmapper.GetGroupVersionResource(restMapper, schema.FromAPIVersionAndKind(resource.GetAPIVersion(), resource.GetKind()))
+	gvr, err := restmapper.GetGroupVersionResource(restMapper, schema.FromAPIVersionAndKind(objRef.APIVersion, objRef.Kind))
 	if err != nil {
-		klog.Errorf("Failed to get GVR from GVK(%s/%s), Error: %v", resource.GetAPIVersion(), resource.GetKind(), err)
+		klog.Errorf("Failed to get GVR from GVK(%s/%s), Error: %v", objRef.APIVersion, objRef.Kind, err)
 		return err
 	}
 
-	if !interpreter.HookEnabled(resource.GroupVersionKind(), configv1alpha1.InterpreterOperationAggregateStatus) {
+	gvk := schema.GroupVersionKind{Group: gvr.Group, Version: gvr.Version, Kind: objRef.Kind}
+	if !interpreter.HookEnabled(gvk, configv1alpha1.InterpreterOperationAggregateStatus) {
 		return nil
 	}
-	newObj, err := interpreter.AggregateStatus(resource, bindingStatus.AggregatedStatus)
-	if err != nil {
-		klog.Errorf("Failed to aggregate status for resource(%s/%s/%s, Error: %v", gvr, resource.GetNamespace(), resource.GetName(), err)
-		return err
-	}
 
-	oldStatus, _, _ := unstructured.NestedFieldNoCopy(resource.Object, "status")
-	newStatus, _, _ := unstructured.NestedFieldNoCopy(newObj.Object, "status")
-	if reflect.DeepEqual(oldStatus, newStatus) {
-		klog.V(3).Infof("Ignore update resource(%s/%s/%s) status as up to date.", gvr, resource.GetNamespace(), resource.GetName())
-		return nil
-	}
+	resource := &unstructured.Unstructured{}
+	if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// get resource template directly from apiserver, rather than informer cache, detail refers to:
+		// https://github.com/karmada-io/karmada/issues/5285
+		resource, err = dynamicClient.Resource(gvr).Namespace(objRef.Namespace).Get(ctx, objRef.Name, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				// It might happen when the resource template has been removed but the garbage collector hasn't removed
+				// the ResourceBinding which dependent on resource template.
+				// So, just return without retry(requeue) would save unnecessary loop.
+				return nil
+			}
+			klog.Errorf("Failed to fetch resource(%s/%s) for resourceBinding. Error: %v.", resource.GetNamespace(), resource.GetName(), err)
```

Comment:
```suggestion
			klog.Errorf("Failed to fetch resource template(%s/%s). Error: %v.", resource.GetNamespace(), resource.GetName(), err)
```

Reply:
In addition, add gvr to the log.

Reply:
ok, this line I finally changed to: 

```go
klog.Errorf("Failed to fetch resource template(%s/%s/%s), Error: %v.", gvr, objRef.Namespace, objRef.Name, err)
```

I found that if err is not nil, that `resource` may be nil, so I use `objRef.Name` to print log.